ReactCommon/fabric/components/text/paragraph/ParagraphShadowNode.cpp

```diff
@@ -38,21 +38,18 @@ void ParagraphShadowNode::setMeasureCache(
   measureCache_ = cache;
 }
 
-void ParagraphShadowNode::updateLocalDataIfNeeded() {
+void ParagraphShadowNode::updateStateIfNeeded() {
   ensureUnsealed();
 
   auto attributedString = getAttributedString();
-  auto currentLocalData =
-      std::static_pointer_cast<const ParagraphLocalData>(getLocalData());
-  if (currentLocalData &&
-      currentLocalData->getAttributedString() == attributedString) {
+  auto state = getStateData();
```

Comment:
It must be `auto const &state = getStateData();` (I added `&`), using a reference here will prevent expensive copying.